### DISCUSSION TITLE
Analyzer warning for email links

### DISCUSSION THREAD

**Discussion by StevenTCramer** [statiqdev/Discussions#170] [Q&A]
```markdown
[email me](mailto:StevenTCramer@gmail.com)
```
Will give the following 
```console
 Invalid URI: <a href="mailto:StevenTCramer&amp;#64;gmail.com">mailto:StevenTCramer&amp;#64;gmail.com</a> (ValidateRelativeLinks)
 ```

How can I have it not try and validate mailto links?

**Comment by daveaglick** [comment]
Ideally it should be skipping `mailto` links so that's something I'll need to take a closer look at. In the meantime, you can turn validation off for any link by adding a `data-no-validate` attribute to the link.

Markdown syntax for attributes is a little weird, so you can either write the following...

```
[Email Me](mailto:foo@bar.com){data-no-validate}
```

Or just write the whole link out longhand directly in the Markdown:

```
<a href="mailto:foo@bar.com" data-no-validate>Email Me</a>
```

That should silence the analyzer for this particular link until I can figure out why it's flagging the `mailto` scheme in the first place (https://github.com/statiqdev/Statiq.Web/issues/998).

**Reply by daveaglick** [reply]
What version of Statiq are you running? This test succeeds which makes me either think you're on an older version and this area has been improved, or something about the link encoding in the generated HTML you provided is throwing the analyzer off.

![image](https://user-images.githubusercontent.com/1020407/191347807-19c013cd-fbf7-49f4-8c86-3e26e716de8d.png)

**Reply by daveaglick** [reply]
Nevermind - I'm able to replicate now on a test project. For some reason the analyzers in isolation seem to be working, but fail with the behavior you saw when run during an actual generation, which is probably why I didn't catch it earlier. Continuing to research.
